File: src/in_memory_store_java/KVStore.java
package in_memory_store_java;

import java.util.HashMap;
import java.util.Map;

public class KVStore implements Store {
    private final Map<String, Entry> data;

    public KVStore() {
        this.data = new HashMap<>();
    }

    public void put(String key, String value) {
        data.put(key, new Entry(value));
    }

    public String get(String key) {
        Entry entry = data.get(key);
        return entry == null ? null : entry.getValue();
    }

    public void delete(String key) {
        data.remove(key);
    }

}

File: in_memory_store/kvStore.go
package main


type Store interface {
	Put(key string, value string)
	Get(key string) (string, bool)
	Delete(key string)
}

type Entry struct {
	Value string
}

type InMemoryStore struct {
	data map[string]Entry
}

func INewInMemoryStore() *InMemoryStore {
	return &InMemoryStore{
		data: make(map[string]Entry),
	}
}

func (kvs *InMemoryStore) Put(key string, value string) {
	kvs.data[key] = Entry{Value: value}
}

func (kvs *InMemoryStore) Get(key string) (string, bool) {
	entry, exists := kvs.data[key]
	if !exists {
		return "", false
	}
	return entry.Value, true
}

func (kvs *InMemoryStore) Delete(key string) {
	delete(kvs.data, key)
}
